yaas/host.py

```diff
@@ -1,4 +1,56 @@
 # coding: utf-8
 
+from __future__ import absolute_import
+from __future__ import print_function
+
+import inspect
+import pprint
+import requests
+import sys
+
+from . import config
+
+
 def command(parser, args):
-    pass
+    """ Interact with registered agents. """
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subcommands = {
+        'list': _list,
+        }
+    for name, fn in subcommands.items():
+        subparsers.add_parser(name, help=inspect.getdoc(fn))
+    subargs, extra = parser.parse_known_args(args)
+    subcommands[subargs.subcommand](
+        subparsers.choices[subargs.subcommand],
+        extra)
+
+
+def _list(parser, args):
+    """ List all registered agents. """
+    parser.add_argument(
+        '--fields',
+        help="Print host details.")
+    subargs, extra = parser.parse_known_args(args)
+    response = requests.get(
+        config.href('/api/v1/hosts'),
+        params={'fields': subargs.fields} if subargs.fields is not None else None,
+        **config.requests_opts())
+    if config.args.raw:
+        pprint.pprint(response.json())
+        sys.exit(0)
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        print(response.json()['message'], file=sys.stderr)
+        sys.exit(1)
+    for item in response.json()['items']:
+        line = []
+        line_keys = ['host_name', 'cluster_name']
+        for key in line_keys:
+            if item['Hosts'].get(key) is not None:
+                line.append(item['Hosts'][key])
+        print(' - '.join(line))
+        for key, value in item['Hosts'].items():
+            if key not in line_keys:
+                print('{indent}{key}: {value}'.format(
+                    indent=' '*4, key=key, value=value))
```

Comment:
`--fields` can only be used with `yaas --raw` then? I'm meh about either solution... Printing random key/value pairs isn't great, but not supporting `--fields` on normal output isn't great either.